packages/cw-derive/src/lib.rs

```diff
@@ -0,0 +1,222 @@
+use convert_case::{Case, Casing};
+use proc_macro::TokenStream;
+use proc_macro2::TokenStream as TokenStream2;
+use quote::quote;
+use syn::fold::Fold;
+use syn::parse::{Parse, Parser};
+use syn::spanned::Spanned;
+use syn::{parse_macro_input, Error, FnArg, Ident, ItemTrait, Pat, TraitItem, TraitItemMethod};
+
+mod parser;
+mod strip_input;
+
+use strip_input::StripInput;
+
+/// Macro generating messages from contract trait.
+///
+/// ## Example usage
+/// ```ignore
+/// # use cosmwasm_std::Response;
+///
+/// # struct Ctx;
+/// # struct Error;
+///
+/// # #[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, schemars::JsonSchema)]
+/// # struct Member;
+///
+/// #[cw_derive::interface(module=msg, exec=Execute, query=Query)]
+/// trait Cw4 {
+///     #[msg(exec)]
+///     fn update_admin(&self, ctx: Ctx, admin: Option<String>) -> Result<Response, Error>;
+///
+///     #[msg(exec)]
+///     fn update_members(&self, ctx: Ctx, remove: Vec<String>, add: Vec<Member>)
+///         -> Result<Response, Error>;
+///
+///     #[msg(query)]
+///     fn admin(&self, ctx: Ctx) -> Result<Response, Error>;
+///
+///     #[msg(query)]
+///     fn member(&self, ctx: Ctx, addr: String, at_height: Option<u64>) -> Result<Response, Error>;
+/// }
+/// ```
+///
+/// This would generate output like:
+///
+/// ```ignore
+/// mod msg {
+///     # #[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, schemars::JsonSchema)]
+///     # struct Member;
+///
+///     #[derive(serde::Serialize, serde::Deserialize, Clone, Debug, PartialEq, schemars::JsonSchema)]
+///     #[serde(rename_all = "snake_case")]
+///     enum Execute {
+///         UpdateAdmin { admin: Option<String> },
+///         UpdateMembers {
+///             remove: Vec<String>,
+///             add: Vec<Member>,
+///         },
+///         AddHook { addr: String },
+///         RemoveHook { addr: String },
+///     }
+/// }
+/// ```
+///
+/// ## Parameters
+///
+/// `interface` attribute takes optional parameters:
```

Comment:
Mostly because of naming collisions. You want to implement many traits on single contract. Probably you want them named differently. And in general - you have more flexibility this way. You can always omit all of those, as they have their defaults.